src/System.Text.Utf8String/System/Text/Utf8Span.cs

```diff
@@ -54,89 +41,47 @@ public Utf8Span(string utf16String)
-        }
+        private Utf8Span(byte[] noCopyUtf8Bytes) => _buffer = noCopyUtf8Bytes;
 
-        private static byte[] CreateArrayFromFieldHandle(RuntimeFieldHandle utf8Data, int length)
-        {
-            var array = new byte[length];
-            RuntimeHelpers.InitializeArray(array, utf8Data);
-            return array;
-        }
+        public static Utf8Span Empty => s_empty;
 
-        public static Utf8Span Empty { get { return s_empty; } }
+        public bool IsEmpty => Bytes.Length == 0;
 
-        /// <summary>
-        /// Returns length of the string in UTF-8 code units (bytes)
-        /// </summary>
-        public int Length => _buffer.Length;
+        public Utf8CodePointEnumerator GetEnumerator() => new Utf8CodePointEnumerator(_buffer);
 
-        public Utf8CodePointEnumerator GetEnumerator()
-        {
-            return new Utf8CodePointEnumerator(_buffer);
-        }
+        public static implicit operator ReadOnlySpan<byte>(Utf8Span utf8Span) => utf8Span.Bytes;
 
-        public static implicit operator ReadOnlySpan<byte>(Utf8Span utf8)
-        {
-            return utf8.Bytes;
-        }
+        public static explicit operator Utf8Span(string utf16String) => new Utf8Span(utf16String);
 
-        public static explicit operator Utf8Span(string s)
-        {
-            return new Utf8Span(s);
-        }
-
-        public static explicit operator string(Utf8Span s)
-        {
-            return s.ToString();
-        }
+        public static explicit operator string(Utf8Span utf8String) => utf8String.ToString();
 
         public ReadOnlySpan<byte> Bytes => _buffer;
 
-        public override string ToString()
-        {
-            var status = Encodings.Utf8.ToUtf16Length(this.Bytes, out int needed);
-            if (status != Buffers.OperationStatus.Done)
-                return string.Empty;
-
-            // UTF-16 is 2 bytes per char
-            var chars = new char[needed >> 1];
-            var utf16 = new Span<char>(chars).AsBytes();
-            status = Encodings.Utf8.ToUtf16(this.Bytes, utf16, out int consumed, out int written);
-            if (status != Buffers.OperationStatus.Done)
-                return string.Empty;
-
-            return new string(chars);
-        }
+        public override string ToString() => Utf8ToUtf16(Bytes);
 
-        public bool ReferenceEquals(Utf8Span other)
-        {
-            return _buffer == other._buffer;
-        }
+        public bool ReferenceEquals(Utf8Span other) => Bytes == other._buffer;
 
-        public bool Equals(Utf8Span other)
-        {
-            return _buffer.SequenceEqual(other._buffer);
-        }
+        public bool Equals(Utf8Span other) => Bytes.SequenceEqual(other._buffer);
+
+        public bool Equals(Utf8String other) => Equals(other.Span);
 
         public bool Equals(string other)
         {
             Utf8CodePointEnumerator thisEnumerator = GetEnumerator();
+            Debug.Assert(BitConverter.IsLittleEndian);
```

Comment:
Why is this assert necessary?

Reply:
If we ever run (and test) in environments where System.String is UTF16BE this assert will fail and will remind us to implement Utf16BECodePointEnumerator